src/BenchmarksApps/TodosApi/DatabaseHealthCheck.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Npgsql;
+
+namespace TodosApi;
+
+public class DatabaseHealthCheck : IHealthCheck
+{
+    private readonly NpgsqlDataSource _dataSource;
+
+    public DatabaseHealthCheck(NpgsqlDataSource dataSource)
+    {
+        _dataSource = dataSource;
+    }
+
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+    {
+        var result = await _dataSource.ExecuteScalarAsync("SELECT COUNT(id) FROM public.todos");
```

Comment:
(nit) `Select top 1` ? Do we really need the total count to ensure the DB is healthy?

Reply:
Nope, but it's indexed so it doesn't really matter. It's really just a nominal check to ensure the table is there. Happy to go with whatever folks think is the best approach for a DB health check here.

Reply:
https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-7.0#database-probe

suggests `select 1`. If we want to ensure the table is there, maybe 

```
SELECT id FROM public.todos LIMIT 1
```

?

Reply:
cc @roji

Reply:
I updated it to simply perform a `LIMIT 1` query against the relevant table

Reply:
Yeah, for pure health checks a `SELECT 1` would be the usual thing, no need to even access any particular table. That's different from a check whether your schema exists, which isn't a periodic health check thing but more of a startup thing.

Probably doesn't matter a great deal in any case here.